lib/libc/Kconfig

```diff
@@ -18,6 +18,12 @@ config REQUIRES_FLOAT_PRINTF
 	  Select a printf implementation that provides a complete
 	  implementation including floating point support.
 
+config ALLOW_NON_64_BIT_TIME_T
```

Comment:
Can we namespace this option with LIBC?

Reply:
> Can we namespace this option with LIBC?

+1 I know the libc configs are currently all over the place (mainly due to things being added ad-hoc by many people over a long time ...); but, we really should stop adding more configs that leak into the global space.

Maybe `ALLOW_LIBC_NON_64_BIT_TIME_T`? Or even just `REQUIRES_LIBC_64_BIT_TIME_T` to align with the rest?

Reply:
I used ALLOW to let other Kconfig files use `select` if necessary; I don't know of a way to do that with REQUIRES without adding another Kconfig variable?

As to the name, adding LIBC seems like a requirement.